src/sdam/server_description.ts

```diff
@@ -135,6 +135,16 @@ class ServerDescription {
     return WRITABLE_SERVER_TYPES.has(this.type);
   }
 
+  get host() {
```

Comment:
This isn't documented anywhere, but one of my goals with the SDAM and CMAP rewrites was to abstract endpoints we could connect to. I prefer names like `address` or `uri` to `host` and `port` because we might eventually want to connect to something that doesn't use a classic host/port (even domain sockets break this, but what if we wanted to connect to.. a file or something). I think adding `host` and `port` on `ServerDescription` is coupling the concerns of creating a TCP socket with something that shouldn't care about transports.

Maybe we can spend a little bit of time thinking about whether such a goal is possible with limited changes: does it make sense for `connect` to check for an `address` property and do the parsing there? Does it make sense to introduce a `ServerAddress` type, and make the above intent more explicit throughout the codebase?

Reply:
I think that's a good idea, it feels a bit brittle to rely on these host and port getters I'll investigate what that would look like. I also have to port the changes to 3.5 and 3.6, but my inclination in 4.0 would be to lean on typescript types to give us a union of possible addresses and use an indicator to differentiate what type of address we'd have. While that can literally translate to javascript is might not feel as assured as when you have type checking. 
But that's just an inclination, I do think this is worthy of breaking out into separate work

Reply:
ok, can you make a ticket to do this work in the future?

Reply:
Yes! https://jira.mongodb.org/browse/NODE-2715